doc/how_to_guides/trajectory_cache/trajectory_cache_tutorial.rst

```diff
@@ -0,0 +1,325 @@
+
+The cache extracts these features from the planning request and plan response, storing them in the cache database.
+When a new planning request is used to attempt to fetch a matching plan, the cache attempts to fuzzily match the request to pre-existing cache entries keyed on similar requests.
+Any "close enough" matches are then returned as valid cache hits for plan reuse, with the definition of "close enough" depending on the type of feature that is being extracted.
+
+For more information about how the cache works and what specific features are used to key cache entries with, please see the :moveit_codedir:`package README <moveit_ros/trajectory_cache/README.md>` and the extensive API documentation for :cpp_api:`moveit_ros::trajectory_cache::TrajectoryCache` and its related classes.
+
+Demo
+----
+
+.. raw:: html
+
+    <video width="700px" controls="true" autoplay="true" loop="true">
+        <source src="../../../_static/videos/trajectory_cache/02_trajectory_cache_demo.webm" type="video/webm">
+        Trajectory cache demo.
+    </video>
+
+(Video is 4x speed.)
+
+This demo has four phases that can be advanced using the ``rviz_visual_tools`` dialog box, which combined, showcases the highlights of the cache's capabilities.
+
+1. Plan and cache (no pruning)
+2. Plan and cache (with pruning)
+3. Fetch from cache and execute (while still planning and caching with pruning)
+4. Fetch from cache and execute, except with large start tolerances
+
+Additionally, the `demo source code itself <how_to_guides/trajectory_cache/src/trajectory_cache_demo.cpp>`_ is heavily annotated with comments and tutorial notes, should you wish to inspect the code and dive deeper.
+
+Pre-Requisites
+++++++++++++++
+
+Make sure you have built the `moveit2_tutorials <https://github.com/moveit/moveit2_tutorials>`_ package and sourced the workspace it is built in to ensure that you can run the demo.
+
+Running the Demo
+++++++++++++++++
+
+In one terminal, start the move group:
+
+::
+
+  ros2 launch moveit2_tutorials trajectory_cache_move_group.launch.py
+
+In another, launch the demo:
+
+::
+
+  ros2 launch moveit2_tutorials trajectory_cache_demo.launch.py
+
+You should see something similar to the video, except with a different set of randomly generated goal poses.
+
+Click ``next`` on the rviz window to advance the demo.
+
+.. note::
+  Sometimes a randomly generated demo goal pose is unreachable (or the ``move_group`` fails to initialize properly).
+  If this happens, the demo will halt in the first phase due to a failed planning call or other issue.
+  
+  Just restart the demo, which will generate new demo goal poses, and resolve the issue.
+
+Configuring the Demo
+++++++++++++++++++++
+
+Additionally, the demo's launch file exposes launch arguments that allows you to change many aspects of the demo.
+Look at the `demo's launch file <how_to_guides/trajectory_cache/launch/trajectory_cache_demo.launch.py>`_ for the full list of configurable launch arguments.
```

Comment:
I think this is failing the `htmlproofer` check because you need to use the `:codedir:` command

Reply:
Sent a change, fingers crossed!